### PR TITLE
Add support for querying DiscoverySetChangeEvent

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

So that it's queryable via public API `GetEventsByAccessPath`.

At the same time, querying a event stream that's not present is no longer an error, rather, we return a empty vec.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Unit tests added.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: types/src/account_state/mod.rs [hunk 1]
```diff
@@ -21,6 +27,31 @@ impl AccountState {
...
+            .get(&*DISCOVERY_SET_RESOURCE_PATH)
+            .map(|bytes| lcs::from_bytes(bytes))
+            .transpose()
+            .map_err(Into::into)
+    }
+
+    pub fn get_event_handle_by_query_path(&self, query_path: &[u8]) -> Result<Option<EventHandle>> {
+        let event_handle = if *ACCOUNT_RECEIVED_EVENT_PATH == query_path {
+            self.get_account_resource()?
+                .map(|account_resourese| account_resourese.received_events().clone())
```
Comment (sblackshear) [inline types/src/account_state/mod.rs]: `resourese` -> `resource` in a couple of places here

File: types/src/account_state/mod.rs [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
+        DiscoverySetResource, DISCOVERY_SET_CHANGE_EVENT_PATH, DISCOVERY_SET_RESOURCE_PATH,
+    },
+    event::EventHandle,
+};
+use anyhow::{bail, Error, Result};
+use serde::{export::Formatter, Deserialize, Serialize};
+use std::{collections::btree_map::BTreeMap, convert::TryFrom, fmt};
 
 #[derive(Default, Deserialize, Serialize)]
 pub struct AccountState(BTreeMap<Vec<u8>, Vec<u8>>);
```
Comment (lightmark) [inline types/src/account_state/mod.rs]: What's the blob on index 1?

File: types/src/account_state/mod.rs [hunk 3]
```diff
@@ -21,6 +27,31 @@ impl AccountState {
...
+                .map(|account_resource| account_resource.received_events().clone())
+        } else if *ACCOUNT_SENT_EVENT_PATH == query_path {
+            self.get_account_resource()?
+                .map(|account_resource| account_resource.sent_events().clone())
+        } else if *DISCOVERY_SET_CHANGE_EVENT_PATH == query_path {
+            self.get_discovery_set_resource()?
+                .map(|discovery_set_resource| discovery_set_resource.change_events().clone())
+        } else {
+            bail!("Unrecognized query path: {:?}", query_path);
+        };
```
Comment (lightmark) [inline types/src/account_state/mod.rs]: why not use `match`?

File: storage/libradb/src/lib.rs [hunk 4]
```diff
@@ -241,20 +239,20 @@ impl LibraDB {
...
-        } else {
-            return Ok((Vec::new(), account_state_with_proof));
+        let event_key = {
+            let (event_key_opt, _count) =
+                account_state_with_proof.get_event_key_and_count_by_query_path(&query_path.path)?;
+            if let Some(event_key) = event_key_opt {
+                event_key
+            } else {
+                return Ok((Vec::new(), account_state_with_proof));
+            }
```
[the remark this answers is not in the document]
Comment (msmouse) [reply storage/libradb/src/lib.rs]: That doesn't quit the method like a return statement does.

File: types/src/account_state/mod.rs (hunk 2 above)
Comment (msmouse) [reply types/src/account_state/mod.rs]: ? 

There is only .0 and it's a BTreeMap.

Comment (lightmark) [reply types/src/account_state/mod.rs]: I need to go to sleep.

File: types/src/account_state/mod.rs (hunk 3 above)
Comment (msmouse) [reply types/src/account_state/mod.rs]: I wasn't able to get the compiler happy with my `match`s... ( you can't even put  `query_path` on the left side of the `==` operator)

Comment (lightmark) [reply types/src/account_state/mod.rs]: This sounds weird to me...

Comment (msmouse) [reply types/src/account_state/mod.rs]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3ac7648a6ec1a4a59dd4b777d460eb6f

Looks like you can't put a variable name as a pattern on match legs, even if it's a static ^

File: storage/libradb/src/libradb_test.rs [hunk 5]
```diff
@@ -454,4 +455,38 @@ proptest! {
         assert!(account_state_with_proof.blob.is_none());
         assert!(events.is_empty());
     }
+
+    #[test]
+    fn test_get_from_non_existent_event_stream(
```
Comment (wqfish) [inline storage/libradb/src/libradb_test.rs]: It feels like this belongs to another commit. If this is about DISCOVERY_SET_CHANGE_EVENT maybe rename a bit.

**Comment by msmouse** [comment]
@r=wqfish
